### PR TITLE
mes-7879-Unit tests

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
Increased code coverage
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/pages/test-report/cat-a-mod1/__tests__/test-report.cat-a-mod1.page.spec.ts [hunk 1]
```diff
@@ -40,10 +57,12 @@ import { CompetencyButtonComponent } from '../../components/competency-button/co
 import { CompetencyComponent } from '../../components/competency/competency';
 import { TestReportCatAMod1Page } from '../test-report.cat-a-mod1.page';
 import { EcoComponent } from '../../components/eco/eco';
+import Expected = jasmine.Expected;
```
Comment (matthew2564) [inline src/app/pages/test-report/cat-a-mod1/__tests__/test-report.cat-a-mod1.page.spec.ts]: Not sure what this is? Cant recall needing this elsewhere

File: src/app/providers/date-time/__tests__/date-time.spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { TestBed } from '@angular/core/testing';
+import { DateTimeProvider } from '@providers/date-time/date-time';
+import { AppConfigProvider } from '@providers/app-config/app-config';
+import { AppConfigProviderMock } from '@providers/app-config/__mocks__/app-config.mock';
+import { DateTime } from '@shared/helpers/date-time';
+
+describe('Compression Provider', () => {
```
Comment (matthew2564) [inline src/app/providers/date-time/__tests__/date-time.spec.ts]: Please update description

File: src/directives/__tests__/character-count.directive.spec.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+class ElementRefMock extends ElementRef {
+  nativeElement = {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    getAttribute(qualifiedName: string): string | null {
+      return '10';
+    },
+  };
+}
+
+describe('Directive: CharCountComponent', () => {
```
Comment (matthew2564) [inline src/directives/__tests__/character-count.directive.spec.ts]: ```describe('Directive: CharacterCountDirective', () => {```

File: src/app/pages/test-report/cat-a-mod1/__tests__/test-report.cat-a-mod1.page.spec.ts (hunk 1 above)
Comment (RLCorp) [reply src/app/pages/test-report/cat-a-mod1/__tests__/test-report.cat-a-mod1.page.spec.ts]: addressed in newest commit

File: src/app/providers/date-time/__tests__/date-time.spec.ts (hunk 2 above)
Comment (RLCorp) [reply src/app/providers/date-time/__tests__/date-time.spec.ts]: addressed in newest commit

File: src/directives/__tests__/character-count.directive.spec.ts (hunk 3 above)
Comment (RLCorp) [reply src/directives/__tests__/character-count.directive.spec.ts]: addressed in newest commit

**Review by matthew2564 — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**